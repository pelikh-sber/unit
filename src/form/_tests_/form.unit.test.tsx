import { render, screen, fireEvent } from "@testing-library/react";
import { Form } from "../";

describe("Форма авторизации", () => {
  it("Рендерится успешно", () => {
    const { baseElement } = render(<Form buttonLabel="some" />);

    expect(baseElement).toMatchSnapshot();

    expect(screen.getByText("some")).toBeInTheDocument();
  });
  it("Заполняется данными и вызывает ивент отправки", async () => {
    const { baseElement } = render(<Form buttonLabel="some" />);

    fireEvent.input(screen.getByTestId("login_input"), {
      target: { value: "login_text" },
    });
    fireEvent.input(screen.getByTestId("password_input"), {
      target: { value: "password_text" },
    });
    const login_input = screen.getByTestId("login_input") as HTMLInputElement;
    const password_input = screen.getByTestId(
      "password_input"
    ) as HTMLInputElement;

    expect(login_input.value).toBe("login_text");
    expect(password_input.value).toBe("password_text");

    expect(baseElement).toMatchSnapshot();
  });
});
