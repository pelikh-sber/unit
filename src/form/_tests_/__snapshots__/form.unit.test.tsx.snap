// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Форма авторизации Заполняется данными и вызывает ивент отправки 1`] = `
<body>
  <div>
    <form>
      <legend>
        Do you agree to the terms?
      </legend>
      <label>
        <input
          data-testid="login_input"
          name="login"
        />
      </label>
      <label>
        <input
          data-testid="password_input"
          name="password"
        />
      </label>
      <button
        data-testid="submit_button"
      >
        some
      </button>
    </form>
  </div>
</body>
`;

exports[`Форма авторизации Рендерится успешно 1`] = `
<body>
  <div>
    <form>
      <legend>
        Do you agree to the terms?
      </legend>
      <label>
        <input
          data-testid="login_input"
          name="login"
        />
      </label>
      <label>
        <input
          data-testid="password_input"
          name="password"
        />
      </label>
      <button
        data-testid="submit_button"
      >
        some
      </button>
    </form>
  </div>
</body>
`;
